fusesoc/signature.py

```diff
@@ -0,0 +1,173 @@
+#!/usr/bin/env python
+# Copyright FuseSoC contributors
+# Licensed under the 2-Clause BSD License, see LICENSE for details.
+# SPDX-License-Identifier: BSD-2-Clause
+
+import base64
+import logging
+import os
+import shutil
+import subprocess
+import sys
+import tempfile
+
+import yaml
+
+from fusesoc import utils
+
+logger = logging.getLogger(__name__)
+
+
+def read_core(data_file_name):
+    """
+    Reads a given core file and returns the canonical (signed)
+    part.
+    """
+    file = open(data_file_name, "rb")
+    core_raw = file.read()
+    file.close()
+    if core_raw.startswith(b"CAPI=2:"):
+        # Core file is single document, no built in signature.  We
+        # sign everything after the header line.
+        core_canonical = core_raw[7:].strip()
+    else:
+        # Core file is not a valid CAPI=2 document.
+        raise RuntimeError("Signature file is not a valid CAPI=2 document.")
+    return core_canonical
+
+
+def sign(data_file_name, key_file_name, old_sig_file):
```

Comment:
I still want the core object to be passed here instead. The actual data file is to me a private detail of the core. We get the name directly by `core.name` and avoid having to read the .core file as a yaml file (which might change in future CAPI versions). The actual data to sign should be fetched from the core through an api call, e.g. `core_canonical = core.get_data_to_sign()`